test/chunkify/chunkify_test.js

```diff
@@ -0,0 +1,125 @@
+     */
+    it('should not throw an error when an empty String is passed in with valid CHAR_LIMIT', function(done) {
+        assert.doesNotThrow(() => {chunkify(EMPTY_STRING, SMALL_CHAR_LIMIT)});
+        done();
+    });
+
+    /**
+     * Tests whether valid message within a specified CHAR_LIMIT is un-modified.
+     */
+
+    it('should not modify valid simple messages', function(done) {
+        const actualChunkifiedMessage = chunkify("Valid Message", TELEGRAM_CHAR_LIMIT);
+        assert.deepEqual(actualChunkifiedMessage, ["Valid Message"]);
+        done();
+    });
+
+    /**
+     * Tests whether small valid messages containing CHAR_LIMIT characters is un-modified.
+     */
+
+    it('should not modify valid small messages', function(done) {
+        const VALID_SMALL_TEST_MESSAGE = randomTextGenerator(SMALL_CHAR_LIMIT);
+        const actualChunkifiedMessage = chunkify(VALID_SMALL_TEST_MESSAGE, SMALL_CHAR_LIMIT);
+        assert.deepEqual(actualChunkifiedMessage, [VALID_SMALL_TEST_MESSAGE]);
+        done();
+    });
+
+    /**
+     * Tests whether large valid messages containing CHAR_LIMIT characters is un-modified.
+     */
+
+    it('should not modify valid large messages', function(done) {
+        const VALID_LARGE_TEST_MESSAGE = randomTextGenerator(TELEGRAM_CHAR_LIMIT);
+        const actualChunkifiedMessage = chunkify(VALID_LARGE_TEST_MESSAGE, TELEGRAM_CHAR_LIMIT);
+        assert.deepEqual(actualChunkifiedMessage, [VALID_LARGE_TEST_MESSAGE]);
+        done();
+    });
+
+    /**
+     * Tests whether large messages containing number of characters greater than CHAR_LIMIT characters are modified appropriately
+     * (i.e. no Empty Strings within output).
+     */
+
+    it('should split large messages in order of linebreaks, whitespace and characters where appropriate', function(done) {
+        const LARGE_TEST_MESSAGE = randomTextGenerator(TELEGRAM_CHAR_LIMIT * Math.floor((Math.random() * 10)));
+        const actualChunkifiedMessage = chunkify(LARGE_TEST_MESSAGE, TELEGRAM_CHAR_LIMIT);
+        for (let i = 0; i < actualChunkifiedMessage.length; ++i){
+            assert(actualChunkifiedMessage[i].length > 0 && actualChunkifiedMessage[i].length <= TELEGRAM_CHAR_LIMIT);
+        }
+        done();
+    });
+
+    /**
+     * Tests whether exception is thrown when invalid/negative CHAR_LIMIT are passed into chunkify
+     */
+
+    it('should throw error when maxChars is less than 1', function(done) {
+        assert.throws(
+            () => {
+                chunkify(EMPTY_STRING, INVALID_CHAR_LIMIT);
+                chunkify(EMPTY_STRING, NEGATIVE_CHAR_LIMIT);
+            }
+        );
+        done();
+    });
+
+    /**
+     * Tests whether small valid messages passed in as text files containing CHAR_LIMIT characters is un-modified.
+     */
+
+    it('should not modify valid small messages passed in as text files', function(done) {
```

Comment:
To me, this implies that we're passing the message as a text file to chunkify

Reply:
Updated documentation for said test case